dev-proxy/ProxyEngine.cs

```diff
@@ -119,12 +118,12 @@ public async Task Run(CancellationToken? cancellationToken)
             _logger.LogInfo($"Listening on {endPoint.IpAddress}:{endPoint.Port}...");
         }
 
-        if (RunTime.IsWindows)
+        if (RunTime.IsWindows && _config.AsSystemProxy)
```

Comment:
Rather than repeating `&& _config.AsSystemProx` on each if statement, let's wrap the whole `if` block in it, because it applies to all cases, including the generic warning at the end, which should only be displayed if you intend to use Dev Proxy as the system proxy.

Reply:
But that would mean that we have either the same condition multiple times, the same else-block twice or missing out on a platform.

```cs
if (_config.AsSystemProxy)
{
    if (RunTime.IsWindows)
    {
        _proxyServer.SetAsSystemHttpsProxy(_explicitEndPoint);
    }
    else if (RunTime.IsMac)
    {
        ToggleSystemProxy(ToggleSystemProxyAction.On, _config.IPAddress, _config.Port);
    }
    else
    {
        _logger.LogWarn("Configure your operating system to use this proxy's port and address");
    }
}
else
{
    _logger.LogWarn("Configure your operating system to use this proxy's port and address");
}
```

```cs
if (RunTime.IsWindows)
{
    if (_config.AsSystemProxy)
    {
        _proxyServer.SetAsSystemHttpsProxy(_explicitEndPoint);
    }
}
else if (RunTime.IsMac)
{
    if (_config.AsSystemProxy)
    {
        ToggleSystemProxy(ToggleSystemProxyAction.On, _config.IPAddress, _config.Port);
    }
}
else
{
    _logger.LogWarn("Configure your operating system to use this proxy's port and address");
}
```

Reply:
Or is it something like this, that you have in mind?
```cs

if (_config.AsSystemProxy)
{
    if (RunTime.IsWindows)
    {
        _proxyServer.SetAsSystemHttpsProxy(_explicitEndPoint);
    }
    else if (RunTime.IsMac)
    {
        ToggleSystemProxy(ToggleSystemProxyAction.On, _config.IPAddress, _config.Port);
    }
    else
    {
        _logger.LogWarn("Configure your operating system to use this proxy's port and address");
    }
}
else
{
    _logger.LogInfo("Configure your application to use this proxy's port and address");
}
```

Reply:
Correct, this latest suggestion is what I had in mind. I like how you added a different info message for configuring the application to use proxy. 👏